master/internal/core.go

```diff
@@ -1053,6 +1053,16 @@ func (m *Master) Run(ctx context.Context) error {
 
 	if m.config.Observability.EnablePrometheus {
 		p := prometheus.NewPrometheus("echo", nil)
+		// Group and obscure URLs returning 400 or 500 errors outside of /api/v1 and /det
+		p.RequestCounterURLLabelMappingFunc = func(c echo.Context) string {
+			if strings.HasPrefix(c.Path(), "/det/") || strings.HasPrefix(c.Path(), "/api/v1/") {
```

Comment:
can you add some really good comments explaining why all of this works the way it did (summarizing our digging).

Reply:
e.g., how the API of RequestCounterURLLabelMappingFunc could be improved to avoid the issue we ran into.